samcli/lib/utils/version_checker.py

```diff
@@ -0,0 +1,126 @@
+"""
+Contains information about newer version checker for SAM CLI
+"""
+import logging
+from datetime import datetime, timedelta
+
+import click
+from requests import get
+from samcli import __version__ as installed_version
+from samcli.cli.global_config import GlobalConfig
+
+LOG = logging.getLogger(__name__)
+
+AWS_SAM_CLI_PYPI_ENDPOINT = "https://pypi.org/pypi/aws-sam-cli/json"
+AWS_SAM_CLI_INSTALL_DOCS = (
+    "https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html"
+)
+PYPI_CALL_TIMEOUT_IN_SECONDS = 5
+DELTA_DAYS = 7
+
+
+def check_newer_version(func):
+    """
+    This function returns a wrapped function definition, which checks if there are newer version of SAM CLI available
+
+    Parameters
+    ----------
+    func: function reference
+        Actual function (command) which will be executed
+
+    Returns
+    -------
+    function reference:
+        A wrapped function reference which executes original function and checks newer version of SAM CLI
+    """
+
+    def wrapped(*args, **kwargs):
+        # execute actual command first
+        actual_result = func(*args, **kwargs)
+        # check and inform newer version if it is available
+        inform_newer_version()
+
+        return actual_result
+
+    return wrapped
+
+
+def inform_newer_version(check_all_always=False):
```

Comment:
I think we usually use the word `force` for this kind of parameter.
Something like: `force_check`

Reply:
Oops that is a type there, will update the param name, thanks!